x-pack/plugins/security_solution/public/sourcerer/experimental/components/dataview_picker/index.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DataViewPicker } from '@kbn/unified-search-plugin/public';
+import React, { useCallback, useRef, useMemo, memo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { useKibana } from '../../../../common/lib/kibana/kibana_react';
+import { DEFAULT_SECURITY_SOLUTION_DATA_VIEW_ID } from '../../constants';
+import { selectDataView } from '../../redux/actions';
+import { sourcererAdapterSelector } from '../../redux/selectors';
+
+const TRIGGER_CONFIG = {
+  label: 'Dataview',
+  color: 'danger',
+  title: 'Experimental data view picker',
+  iconType: 'beaker',
+} as const;
+
+export const DataviewPicker = memo(() => {
+  const dispatch = useDispatch();
+
+  const {
+    services: { dataViewEditor, data, dataViewFieldEditor },
+  } = useKibana();
+
+  const closeDataViewEditor = useRef<() => void | undefined>();
+  const closeFieldEditor = useRef<() => void | undefined>();
+
+  // TODO: should this be implemented like that? If yes, we need to source dataView somehow or implement the same thing based on the existing state value.
+  // const canEditDataView =
+  // Boolean(dataViewEditor?.userPermissions.editDataView()) || !dataView.isPersisted();
+  const canEditDataView = true;
+
+  const { dataViewId } = useSelector(sourcererAdapterSelector);
+
+  const createNewDataView = useCallback(() => {
+    closeDataViewEditor.current = dataViewEditor.openEditor({
+      // eslint-disable-next-line no-console
+      onSave: () => console.log('new data view saved'),
+      allowAdHocDataView: true,
+    });
+  }, [dataViewEditor]);
+
+  const onFieldEdited = useCallback(() => {}, []);
+
+  const editField = useMemo(() => {
+    if (!canEditDataView) {
+      return;
+    }
+    return async (fieldName?: string, _uiAction: 'edit' | 'add' = 'edit') => {
+      if (!dataViewId) {
+        return;
+      }
+
+      const dataViewInstance = await data.dataViews.get(dataViewId);
+      closeFieldEditor.current = dataViewFieldEditor.openEditor({
+        ctx: {
+          dataView: dataViewInstance,
+        },
+        fieldName,
+        onSave: async () => {
+          onFieldEdited();
+        },
+      });
+    };
+  }, [canEditDataView, dataViewId, data.dataViews, dataViewFieldEditor, onFieldEdited]);
+
+  const addField = useMemo(
+    () => (canEditDataView && editField ? () => editField(undefined, 'add') : undefined),
+    [editField, canEditDataView]
+  );
```

Comment:
This part is a bit confusing, why do we need `editField` only to be called with `editField(undefined, 'add')` inside `addField`? Would it make sense to simplify everything into something like this?

```
const addField = useCallback(() => {
  if (!canEditDataView || !dataViewId) {
    return;
  }
  data.dataViews.get(dataViewId).then((dataViewInstance) => {
    closeFieldEditor.current = dataViewFieldEditor.openEditor({
      ctx: { dataView: dataViewInstance },
      fieldName,
      onSave: async () => {
        // TODO
      },
    });
  });
}, [canEditDataView, dataViewId, data.dataViews, dataViewFieldEditor, onFieldEdited]);
```

Reply:
just copied it from the discover, I will get to it in due time:)